spec/lib/jobs/obm-control-spec.js

```diff
@@ -131,6 +132,63 @@ describe("Job.Obm.Node", function () {
             });
         });
 
+        it('should fail if there is more then one OBM service', function(done){
+          var node = {
+              obmSettings: [
+                  {
+                      service: 'noop-obm-service',
+                      config: {}
+                  },
+                  {
+                      service: 'foo',
+                      config: {
+                          "user": "admin",
+                          "password": "admin",
+                          "host": "10.0.0.254"
+                      }
+                  }
+              ]
+          };
+          mockWaterline.nodes.findByIdentifier.resolves(node);
+
+          job.run()
+          .then(function() {
+              done(new Error("Expected job to fail"));
+          })
+          .catch(function(e) {
+              try {
+                  expect(e).to.have.property('name').that.equals('AssertionError');
+                  expect(e).to.have.property('message').that.equals(
+                      'more than one OBM service assigned to this node,' +
+                      ' no default set for which to use'
+                  );
+                  done();
+              } catch (e) {
+                  done(e);
+              }
+          });
+        });
+
+/////////// MY CODE
+        it('should set default OBM setting if only one exists', function(){
+            var node = {
+                obmSettings: [
+                    {
+                        service: 'noop-obm-service',
+                        config: {}
+                    }
+                ]
+            };
+            mockWaterline.nodes.findByIdentifier.resolves(node);
+
+            return job.run()
+            .then(function() {
+                // hmmm, how to verify settings has been set...
+                expect(job.settings).to.equal(node.obmSettings[0]);
+            });
```

Comment:
No a bug here, just to reply to your comment:

Previous, we tested whether the correct obm service has been set by judging whether the correct obmServiceFatory function is called, the code snippt is as following:
```javascript
            return job.run()
            .then(function() {
                var ipmiObmServiceFactory = helper.injector.get('ipmi-obm-service');
                expect(ObmServiceSpy).to.have.been.calledWith(
                    job.nodeId, ipmiObmServiceFactory, node.obmSettings[0],
                    testOptions);
            });
```

Of course, you choose to expose the result obm setting to job's prototype is an alternative simpler way.